crates/cdk-ffi/src/wallet.rs

```diff
@@ -0,0 +1,293 @@
+        include_fees: bool,
+    ) -> Result<Option<Proofs>, FfiError> {
+        let cdk_proofs: Vec<cdk::nuts::Proof> =
+            input_proofs.into_iter().map(|p| p.inner.clone()).collect();
+
+        // Convert spending conditions if provided
+        let conditions = spending_conditions.map(|sc| sc.try_into()).transpose()?;
+
+        let result = self
+            .inner
+            .swap(
+                amount.map(Into::into),
+                amount_split_target.into(),
+                cdk_proofs,
+                conditions,
+                include_fees,
+            )
+            .await?;
+
+        Ok(result.map(|proofs| {
+            proofs
+                .into_iter()
+                .map(|p| std::sync::Arc::new(p.into()))
+                .collect()
+        }))
+    }
+
+    /// Get proofs by states
+    pub async fn get_proofs_by_states(&self, states: Vec<ProofState>) -> Result<Proofs, FfiError> {
+        let mut all_proofs = Vec::new();
+
+        for state in states {
+            let proofs = match state {
+                ProofState::Unspent => self.inner.get_unspent_proofs().await?,
+                ProofState::Pending => self.inner.get_pending_proofs().await?,
+                ProofState::Reserved => self.inner.get_reserved_proofs().await?,
+                ProofState::PendingSpent => self.inner.get_pending_spent_proofs().await?,
+                ProofState::Spent => {
+                    // CDK doesn't have a method to get spent proofs directly
+                    // They are removed from the database when spent
+                    continue;
+                }
+            };
+
+            for proof in proofs {
+                all_proofs.push(std::sync::Arc::new(proof.into()));
+            }
+        }
+
+        Ok(all_proofs)
+    }
+
+    /// Check if proofs are spent
+    pub async fn check_proofs_spent(&self, proofs: Proofs) -> Result<Vec<bool>, FfiError> {
+        let cdk_proofs: Vec<cdk::nuts::Proof> =
+            proofs.into_iter().map(|p| p.inner.clone()).collect();
+
+        let proof_states = self.inner.check_proofs_spent(cdk_proofs).await?;
+        // Convert ProofState to bool (spent = true, unspent = false)
+        let spent_bools = proof_states
+            .into_iter()
+            .map(|proof_state| {
+                matches!(
+                    proof_state.state,
+                    cdk::nuts::State::Spent | cdk::nuts::State::PendingSpent
+                )
+            })
+            .collect();
+        Ok(spent_bools)
+    }
+}
+
+/// Configuration for creating wallets
+#[derive(Debug, Default, Clone, uniffi::Record)]
+pub struct WalletConfig {
+    pub database_path: Option<String>,
+    pub target_proof_count: Option<u32>,
+}
+
+/// Utility functions
+#[uniffi::export]
+pub fn generate_seed() -> Vec<u8> {
+    use rand::RngCore;
+    let mut seed = [0u8; 32];
+    rand::rng().fill_bytes(&mut seed);
+    seed.to_vec()
+}
```

Comment:
Think we should do a mnemonic here, maybe even add a fn so the user can get it after it is generated.